a1/src/main/java/ro/utcn/is/a1/repository/QuestionRepositoryFactory.java

```diff
@@ -0,0 +1,9 @@
+package ro.utcn.is.a1.repository;
+
+
+
+public interface QuestionRepositoryFactory {
```

Comment:
You could have actually created a single "RepositoryFactory", with methods for each type of repository (e.g. `createQuestionRepository`, `createAnswerRepository`, etc). That would have reduced the amount of classes in the persistence packages a lot.